util/chunk/chunk.go

```diff
@@ -277,6 +278,63 @@ func (c *Chunk) AppendPartialRow(colIdx int, row Row) {
 	}
 }
 
+// PreAlloc4Row pre-allocates the memory space for a Row.
+// Nothing except for the nullBitMap of c.columns will be pre-written.
+func (c *Chunk) PreAlloc4Row(row Row) {
+	for i, srcCol := range row.c.columns {
+		dstCol := c.columns[i]
+		dstCol.appendNullBitmap(!srcCol.isNull(row.idx))
+		elemLen := len(srcCol.elemBuf)
+		if !srcCol.isFixed() {
+			elemLen = int(srcCol.offsets[row.idx+1] - srcCol.offsets[row.idx])
+			dstCol.offsets = append(dstCol.offsets, int32(len(dstCol.data)+elemLen))
+		}
+		dstCol.length++
+		needCap := len(dstCol.data) + elemLen
```

Comment:
Could we do not do the following if the value is null?

Reply:
The behavior of this function refers AppendPartialRow,
it seems we still need it even the value is null.